webui/react/src/stores/projects.tsx

```diff
@@ -1,14 +1,17 @@
 import { Loadable, Loaded, NotLoaded } from 'hew/utils/loadable';
 import { Map } from 'immutable';
+import { find, remove } from 'lodash';
 
 import { getWorkspaceProjects } from 'services/api';
+import { Streamable, StreamContent } from 'services/stream';
+import { StreamSubscriber } from 'stores/stream';
 import { Project } from 'types';
 import handleError from 'utils/error';
-import { Observable, observable, WritableObservable } from 'utils/observable';
+import { ImmutableObservable, Observable, observable } from 'utils/observable';
 
-class ProjectStore {
-  #projects: WritableObservable<Map<number, Project>> = observable(Map());
-  #projectsByWorkspace: WritableObservable<Map<string, Project[]>> = observable(Map());
+class ProjectStore implements StreamSubscriber {
+  #projects: ImmutableObservable<Map<number, Project>> = observable(Map());
```

Comment:
`ImmutableObserver` assumes all values under the observable are [value objects](https://immutable-js.com/docs/latest@main/ValueObject/). here, the keys are, but the values aren't. you can make a `Project[]` a value object by using the immutable list object instead of a plain array and wrapping the project objects in an `asValueObject` wrapper.